app/notifications/notifications_ses_callback.py

```diff
@@ -93,8 +96,10 @@ def remove_emails_from_bounce(bounce_dict):
         recip.pop('emailAddress')
 
 
-def _check_and_queue_callback_task(notification_id, service_id):
+def _check_and_queue_callback_task(notification):
     # queue callback task only if the service_callback_api exists
-    service_callback_api = get_service_callback_api_for_service(service_id=service_id)
+    service_callback_api = get_service_callback_api_for_service(service_id=notification.service_id)
     if service_callback_api:
-        send_delivery_status_to_service.apply_async([str(notification_id)], queue=QueueNames.CALLBACKS)
+        encrypted_notification = create_encrypted_callback_data(notification, service_callback_api)
+        send_delivery_status_to_service.apply_async([str(notification.id), encrypted_notification],
```

Comment:
Do we still need notification.id here?

Reply:
You also need to change the definition of send_delivery_status_to_service here:
https://github.com/alphagov/notifications-api/blob/master/app/celery/service_callback_tasks.py#L27

Reply:
I need it for backward compatibility.